src/NzbDrone.Core.Test/MediaFiles/EpisodeImport/ImportDecisionMakerFixture.cs

```diff
@@ -226,10 +233,100 @@ public void should_not_throw_if_episodes_are_not_found()
+        }
+
+        [Test]
+        public void should_not_use_folder_for_full_season()
+        {
+            var videoFiles = new[]
+                             {
+                                 @"C:\Test\Unsorted\Series.Title.S01\S01E01.mkv".AsOsAgnostic(),
+                                 @"C:\Test\Unsorted\Series.Title.S01\S01E02.mkv".AsOsAgnostic(),
+                                 @"C:\Test\Unsorted\Series.Title.S01\S01E03.mkv".AsOsAgnostic()
+                             };
+
+            GivenSpecifications(_pass1);
+            GivenVideoFiles(videoFiles.ToList());
+
+            var folderInfo = Parser.Parser.ParseTitle("Series.Title.S01");
+
+            Subject.GetImportDecisions(_videoFiles, _series, true, folderInfo);
+
+            Mocker.GetMock<IParsingService>()
+                  .Verify(c => c.GetLocalEpisode(It.IsAny<string>(), It.IsAny<Series>(), true, null), Times.Exactly(3));
+
+            Mocker.GetMock<IParsingService>()
+                  .Verify(c => c.GetLocalEpisode(It.IsAny<string>(), It.IsAny<Series>(), true, It.Is<ParsedEpisodeInfo>(p => p != null)), Times.Never());
+        }
+
+        [Test]
+        public void should_not_use_folder_when_it_contains_more_than_one_valid_video_file()
+        {
+            var videoFiles = new[]
+                             {
+                                 @"C:\Test\Unsorted\Series.Title.S01E01\S01E01.mkv".AsOsAgnostic(),
+                                 @"C:\Test\Unsorted\Series.Title.S01E01\1x01.mkv".AsOsAgnostic()
+                             };
+
+            GivenSpecifications(_pass1);
+            GivenVideoFiles(videoFiles.ToList());
+
+            var folderInfo = Parser.Parser.ParseTitle("Series.Title.S01E01");
+
+            Subject.GetImportDecisions(_videoFiles, _series, true, folderInfo);
+
+            Mocker.GetMock<IParsingService>()
+                  .Verify(c => c.GetLocalEpisode(It.IsAny<string>(), It.IsAny<Series>(), true, null), Times.Exactly(2));
+
+            Mocker.GetMock<IParsingService>()
+                  .Verify(c => c.GetLocalEpisode(It.IsAny<string>(), It.IsAny<Series>(), true, It.Is<ParsedEpisodeInfo>(p => p != null)), Times.Never());
+        }
+
+        [Test]
+        public void should_use_folder_when_only_one_video_file()
+        {
+            GivenSpecifications(_pass1);
+
+            var folderInfo = Parser.Parser.ParseTitle("Series.Title.S01E01");
+
+            Subject.GetImportDecisions(_videoFiles, _series, true, folderInfo);
+
+            Mocker.GetMock<IParsingService>()
+                  .Verify(c => c.GetLocalEpisode(It.IsAny<string>(), It.IsAny<Series>(), true, It.IsAny<ParsedEpisodeInfo>()), Times.Exactly(1));
+
+            Mocker.GetMock<IParsingService>()
+                  .Verify(c => c.GetLocalEpisode(It.IsAny<string>(), It.IsAny<Series>(), true, null), Times.Never());
+        }
+
+        [Test]
+        public void should_use_folder_when_only_one_video_file_and_a_sample()
```

Comment:
That means we're missing out on any dir fix releases, exactly the scenario you saw with The Americans.